assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidget.js

```diff
@@ -76,22 +76,45 @@ function DashboardTopEarningPagesWidget( { Widget, WidgetReportZero, WidgetRepor
 		};
 	} );
 
+	const Footer = () => (
+		<SourceLink
+			className="googlesitekit-data-block__source"
+			name={ _x( 'Analytics', 'Service name', 'google-site-kit' ) }
+			href={ analyticsMainURL }
+			external
+		/>
+	);
+
 	if ( loading ) {
-		return <PreviewTable rows={ 5 } padding />;
+		return (
+			<Widget noPadding Footer={ Footer }>
+				<PreviewTable rows={ 5 } padding />
+			</Widget>
+		);
 	}
 
 	// A restricted metrics error will cause this value to change in the resolver
 	// so this check should happen before an error, which is only relevant if they are linked.
 	if ( ! isAdSenseLinked ) {
-		return <AdSenseLinkCTA />;
+		return (
+			<Widget Footer={ Footer }>
+				<AdSenseLinkCTA />
+			</Widget>
+		);
 	}
 
 	if ( error ) {
-		return <WidgetReportError moduleSlug="analytics" error={ error } />;
+		return (
+			<Widget Footer={ Footer } >
+				<WidgetReportError moduleSlug="analytics" error={ error } />
+			</Widget>
+		);
 	}
 
 	if ( isZeroReport( data ) ) {
-		return <WidgetReportZero moduleSlug="analytics" />;
+		<Widget Footer={ Footer }>
+			 <WidgetReportZero moduleSlug="analytics" />
+		</Widget>;
```

Comment:
Don't forget to return this :stuck_out_tongue_closed_eyes:

Reply:
😂  Oh dear. I'm not having a good day am I. 🙈